gentoo_update/shell_runner.py

```diff
@@ -183,7 +221,10 @@ def __del__(self) -> None:
         """
         Closed all file handlers after ShellRunner is closed.
         """
-        if self.logger:
-            for handler in self.logger.handlers:
-                handler.close()
-                self.logger.removeHandler(handler)
+        try:
+            if self.logger:
+                for handler in self.logger.handlers:
+                    handler.close()
```

Comment:
You probably want to catch exceptions in this loop so that you close other handlers if the first one fails, etc.